### PR TITLE
Move terraformer library from gardener to gardener-extensions

### PR DESCRIPTION
**Pull request by tim-ebert**
**What this PR does / why we need it**:
This PR moves the terraformer library package from `gardener/gardener/pkg/operation/terraformer` to `gardener/gardener-extensions/pkg/terraformer`.
Gardener itself does no longer use it but only the extensions, hence, its correct home is in the gardener-extensions library.
It also merges the terraformer wrapper package (`gardener/gardener-extensions/pkg/gardener/terraformer`) with the moved package.
All extension controllers were adapted to only use the newly exposed `gardener/gardener-extensions/pkg/terraformer.Terraformer` interface.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
See also gardener/gardener#1596

**Release note**:

```noteworthy developer
The `terraformer` package was moved from `gardener/gardener` to `gardener/gardener-extensions/pkg/terraformer`.
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
There are some `golangci-lint` failures. Could you please check them?

**Comment by tim-ebert** [comment]
Lint failures are fixed, PTAL. 👍

File: pkg/terraformer/types.go [hunk 1]
```diff
@@ -82,3 +83,30 @@ const (
 	// TerraformerPurposeBackup is a constant for the complete Terraform setup with purpose 'etcd backup'.
 	TerraformerPurposeBackup = "backup"
```
Comment (ialidzhikov) [inline pkg/terraformer/types.go]:
```suggestion

```

File: pkg/terraformer/types.go [hunk 2]
```diff
@@ -15,14 +15,15 @@
 package terraformer
 
 import (
+	"k8s.io/client-go/rest"
```
Comment (ialidzhikov) [inline pkg/terraformer/types.go]: You could move it below.

File: pkg/terraformer/types.go [hunk 3]
```diff
@@ -82,3 +83,30 @@ const (
...
+	SetDeadlinePod(time.Duration) Terraformer
+	SetDeadlineJob(time.Duration) Terraformer
+	InitializeWith(initializer Initializer) Terraformer
+	Apply() error
+	Destroy() error
+	GetStateOutputVariables(variables ...string) (map[string]string, error)
+	ConfigExists() (bool, error)
+}
+
+// Initializer can initialize an Terraformer.
```
Comment (ialidzhikov) [inline pkg/terraformer/types.go]:
```suggestion
// Initializer can initialize a Terraformer.
```

File: pkg/terraformer/types.go (hunk 1 above)
Comment (tim-ebert) [reply pkg/terraformer/types.go]: const removed 👍

**Comment by tim-ebert** [comment]
Thanks for your reviews!
Suggestions applied and rebased onto master, PTAL.

**Pull request merged**